extensions/agroal/deployment/src/main/java/io/quarkus/agroal/deployment/AgroalProcessor.java

```diff
@@ -233,11 +234,15 @@ void generateDataSourceSupportBean(AgroalRecorder recorder,
             return;
         }
 
-        // make a DataSourceProducer bean
+        // make a DataSources bean
         additionalBeans.produce(AdditionalBeanBuildItem.builder().addBeanClasses(DataSources.class).setUnremovable()
                 .setDefaultScope(DotNames.SINGLETON).build());
         // add the @DataSource class otherwise it won't be registered as a qualifier
         additionalBeans.produce(AdditionalBeanBuildItem.builder().addBeanClass(DataSource.class).build());
+        // make source datasources are initialized at startup
+        additionalBeans.produce(AdditionalBeanBuildItem.builder().addBeanClasses(AgroalDataSourcesInitializer.class)
+                .setUnremovable()
+                .setDefaultScope(DotNames.SINGLETON).build());
```

Comment:
Why this default scope? I'd keep `@Dependent` (class with no scope but an observer method) so that the bean is destroyed right after the notification.